util/xxhash3/README.md

```diff
@@ -0,0 +1,126 @@
+# XXH3 hash algorithm
+A Go implementation of the 64/128 bit xxh3 algorithm, added the SIMD vector instruction set: AVX2 and SSE2 support to accelerate the hash processing.\
+The original repository can be found here: [https://github.com/Cyan4973/xxHash].
+
+
+## Overview
+
+For the input length larger than 240, the 64-bit version of xxh3 algorithm goes along with following steps to get the hash result.
+
+### step1.  Initialize 8 accumulators used to store the middle result of each iterator.
+```
+xacc[0] = prime32_3
+xacc[1] = prime64_1
+xacc[2] = prime64_2
+xacc[3] = prime64_3
+xacc[4] = prime64_4
+xacc[5] = prime32_2
+xacc[6] = prime64_5
+xacc[7] = prime32_1
+```
+
+### step2.  Process 1024 bytes of input data as one block each time
+```
+while remaining_length > 1024{
+    for i:=0, j:=0; i < 1024; i += 64, j+=8 {
+        for n:=0; n<8; n++{
+            inputN := input[i+8*n:i+8*n+8]
+            secretN := inputN ^ secret[j+8*n:j+8*n+8]
+            
+            xacc[n^1] += inputN
+            xacc[n]   +=  (secretN & 0xFFFFFFFF) * (secretN >> 32)
+        }
+    }
+    
+    xacc[n]   ^= xacc[n] >> 47
+    xacc[n]   ^= secret[128+8*n:128+8*n:+8]
+    xacc[n]   *= prime32_1
+    
+    remaining_length -= 1024
+}
+```
+
+### step3.  Process remaining stripes (totally 1024 bytes at most)
+```
+
+for i:=0, j:=0; i < remaining_length; i += 64, j+=8 {
+    for n:=0; n<8; n++{
+        inputN := input[i+8*n:i+8*n+8]
+        secretN := inputN ^ secret[j+8*n:j+8*n+8]
+    
+        xacc[n^1] += inputN
+        xacc[n]   += (secretN & 0xFFFFFFFF) * (secretN >> 32)
+    }
+
+    remaining_length -= 64
+}
+```
+
+### step4.  Process last stripe  (align to last 64 bytes)
+```
+for n:=0; n<8; n++{
+    inputN := input[(length-64): (length-64)+8]
+    secretN := inputN ^ secret[121+8*n, 121+8*n+8]
+
+    xacc[n^1] += inputN
+    xacc[n]   += (secretN & 0xFFFFFFFF) * (secretN >> 32)
+}
+```
+
+### step5.  Merge & Avalanche accumulators
+```
+acc = length * prime64_1
+acc += mix(xacc[0]^secret11, xacc[1]^secret19) + mix(xacc[2]^secret27, xacc[3]^secret35) +
+    mix(xacc[4]^secret43, xacc[5]^secret51) + mix(xacc[6]^secret59, xacc[7]^secret67)
+
+acc ^= acc >> 37
+acc *= 0x165667919e3779f9
+acc ^= acc >> 32
+```
+
+If the input length is smaller or equal than 240 bytes, the calculating steps are similar to the above description. The major difference lies in the data alignment. in the case of smaller input, the alignment size is 16 bytes. 
```

Comment:
1) why not `smaller or equal to`
2) the unit is `bytes`, so the `input length` better be `input data size`?
3) `in the case of smaller input`, the first letter should be capitalized

Reply:
fixed